### PR TITLE
Package artist for building directly from NPM

### PR DESCRIPTION
**Pull request by Hamms**
a la https://github.com/code-dot-org/maze/pull/7

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -15,11 +17,14 @@
...
   "devDependencies": {
+    "babel-loader": "^7.1.4",
+    "babel-preset-env": "^1.7.0",
     "canvas": "^1.6.10",
     "codecov": "^3.0.0",
     "file-loader": "^1.1.6",
     "jest": "^22.4.3",
-    "webpack": "^3.10.0"
+    "webpack": "^4.10.2",
+    "webpack-cli": "^2.1.5"
```
[the remark this answers is not in the document]
Comment (Hamms) [reply package.json]: It is; webpack 4 breaks the cli out into its own package. Two packages, actually, so we could use the lighter-weight `webpack-command` instead, I just haven't devoted time to figuring out the pros and cons of that decision. See https://webpack.js.org/guides/installation/#local-installation

**Comment by joshlory** [comment]
I think this might be missing the `yarn.lock` changes.

**Comment by Hamms** [comment]
Ahh, dang, I had been building this with just `npm`, in which it builds just fine.

Looks like the webpack 4 node dependency was made more specific than necessary as per https://github.com/webpack/webpack/issues/6579, but the fix is dependent on a new version of https://github.com/webpack/enhanced-resolve getting released, which might not happen anytime soon.

So it seems like our options are:
1. Go back to webpack 3
2. Update node requirement to be >= 6.11.5
3. Install using `npm` rather than `yarn`

Thoughts?

**Comment by joshlory** [comment]
That's super annoying 😕, it looks like they're just waiting for a `enhanced-resolve` version bump.  No strong preference between (2) and (3).  Are there any downsides to updating our node requirement?

**Comment by Hamms** [comment]
I think the biggest downside is that we then require developers to manage multiple local versions of node being installed if they want to work on artist. I'd lean toward (3) for that reason, and because our dependencies here are lightweight enough that dropping yarn isn't particularly scary.

**Comment by joshlory** [comment]
SGTM!

**Comment by Hamms** [comment]
That's new in npm 5, right? I don't have a strong opinion on that for this project. On the one hand, it seems valuable to do in general. On the other hand, we're not officially on npm 5 for our main code-dot-org project and I'd rather keep the dev environment requirements aligned between the two as much as possible.

**Pull request merged**